hail/python/test/hailtop/inter_cloud/test_fs.py

```diff
@@ -175,7 +225,29 @@ async def test_read_range(filesystem: Tuple[asyncio.Semaphore, AsyncFS, str]):
     except UnexpectedEOFError:
         pass
     else:
-        assert False
+       assert False
+
+
+@pytest.mark.asyncio
+async def test_read_range_end_exclusive_empty_file(filesystem: Tuple[asyncio.Semaphore, AsyncFS, str]):
+    sema, fs, base = filesystem
+
+    file = f'{base}foo'
+
+    await fs.write(file, b'')
+
+    assert await fs.read_range(file, 0, 0, end_inclusive=False) == b''
```

Comment:
What happens if `end_inclusive=True`? Error?

Reply:
That will try to read the zeroth byte from an empty file and will trigger an error. I've added a new test for that case, `test_read_range_end_inclusive_empty_file_should_error`. In Google, you get
```
hailtop.httpx.ClientResponseError: 416, message='Requested range not satisfiable'
```
In local, you get:
```
hailtop.aiotools.fs.exceptions.UnexpectedEOFError
```

Reply:
Hmm. I suppose we should catch and convert these all to UnexpectedEOF errors

Reply:
OK, all now UnexpectedEOFError and a test for that.